### PR TITLE
Not equal

### PR DESCRIPTION
**Pull request by jz543fm**

### CODE REVIEW

File: src/Parser/Language.hs [hunk 1]
```diff
@@ -12,9 +12,11 @@ reservedNames =
     "False",
     "if",
     "else",
+    "else if",
```
Comment (yurtsiv) [inline src/Parser/Language.hs:15]: Unnecessary change

File: src/Parser/Language.hs [hunk 2]
```diff
@@ -12,9 +12,11 @@ reservedNames =
...
     "do",
     "end",
-    "then"
+    "then",
+    "while"
```
Comment (yurtsiv) [inline src/Parser/Language.hs:19]: Unnecessary change

File: src/Parser/Parser.hs [hunk 3]
```diff
@@ -98,6 +98,8 @@ parseBlock = do
    reserved "end"
    return $ Block exprs
 
+
+
```
Comment (yurtsiv) [inline src/Parser/Parser.hs:102]: Unnecessary new lines

File: src/Parser/Parser.hs [hunk 4]
```diff
@@ -138,6 +140,7 @@ operatorsTable =
      ]
    , [ prefixOp "!" (Inverse)
      , binaryOp "||" (Or) AssocRight
+     , binaryOp "/=" (NotEq) AssocRight
```
Comment (yurtsiv) [inline src/Parser/Parser.hs:143]: This should be in the same group as `==` operator

File: src/Parser/Language.hs [hunk 5]
```diff
@@ -31,7 +33,8 @@ reservedOpNames =
     "!",
     "||",
     "&&",
-    "++"
+    "++",
+    "/="
```
Comment (yurtsiv) [inline src/Parser/Language.hs:37]: Place this next to `==`, please

File: test/Interpreter/EvalSpec.hs [hunk 6]
```diff
@@ -41,9 +41,54 @@ spec = do
     testProgramm "1 < 3" (Bool True)
     testProgramm "3 <= 3" (Bool True)
     testProgramm "3 == 2" (Bool False)
+    testProgramm "9 /= 10" (Bool True)
+    testProgramm "4 /= 6" (Bool True)
+    testProgramm "True /= 1" (Bool False)
```
Comment (yurtsiv) [inline test/Interpreter/EvalSpec.hs:46]: When testing your changes and those changes are correct, tests should be green.
`testProgramm` currently asserts only positive cases (we'll add tests for negative cases soon https://github.com/yurtsiv/mooncake/issues/3). So do not test negative cases yet and also keep in mind, that there're two failing tests (as the functionality is not implemented yet) that I forgot to remove:

image

Leave them as-is, we'll handle that in this issue https://github.com/yurtsiv/mooncake/issues/9

Thanks!

File: test/Interpreter/EvalSpec.hs [hunk 7]
```diff
@@ -41,9 +41,54 @@ spec = do
...
+    |]
+
+    testProgramm condZ (Bool True)
+
+
+    let condX = [r|
+      if 0 /= 2 then
+        True
+      end
+    |]
```
Comment (yurtsiv) [inline test/Interpreter/EvalSpec.hs:88]: It's enough to  test the `/=` operator alone, `if` conditions are already covered

**Comment by yurtsiv** [comment]
Closing this since the functionality is already implemented here https://github.com/yurtsiv/mooncake/pull/6

But anyway, thank you for the effort @jz543fm , I appreciate it!
